src/zabapgit_repo_impl.prog.abap

```diff
@@ -401,16 +410,32 @@ CLASS lcl_repo IMPLEMENTATION.
     <ls_return>-file-sha1     = lcl_hash=>sha1( iv_type = gc_type-blob
                                                 iv_data = <ls_return>-file-data ).
 
+    lt_cache = mt_local.
```

Comment:
looks like variable `lt_cache` is not needed? lt_cache and mt_local are identical until line 454

perhaps the type of mt_local needs to be changed to SORTED TABLE